Update Node.lookupNamespaceURI() and XPathEvaluatorBase.createNSResolver()

cdumez opened the pull request:
#### b0e1a9c1d9b3b952faa97d997c57d6358d11b79f
<pre>
Update Node.lookupNamespaceURI() and XPathEvaluatorBase.createNSResolver()
<a href="https://bugs.webkit.org/show_bug.cgi?id=252716">https://bugs.webkit.org/show_bug.cgi?id=252716</a>

Reviewed by Darin Adler.

Update Node.lookupNamespaceURI() and XPathEvaluatorBase.createNSResolver() per:
<a href="https://github.com/whatwg/dom/pull/1165">https://github.com/whatwg/dom/pull/1165</a>
<a href="https://github.com/web-platform-tests/wpt/pull/38636">https://github.com/web-platform-tests/wpt/pull/38636</a>

- lookupNamespaceURI():
Element should handle &quot;xml&quot; and &quot;xmlns&quot; by default.

- createNSResolver():
It should return the argument as is, and should not add &quot;xml&quot; prefix support.

* LayoutTests/imported/w3c/web-platform-tests/dom/nodes/Node-lookupNamespaceURI-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/nodes/Node-lookupNamespaceURI.html:
* LayoutTests/imported/w3c/web-platform-tests/domxpath/xpathevaluatorbase-creatensresolver-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/domxpath/xpathevaluatorbase-creatensresolver.html: Added.
* Source/WebCore/dom/Document.h:
(WebCore::Document::createNSResolverForBindings const):
* Source/WebCore/dom/Node.cpp:
(WebCore::locateDefaultNamespace):
* Source/WebCore/xml/XPathEvaluator.h:
(WebCore::XPathEvaluator::createNSResolverForBindings const):
* Source/WebCore/xml/XPathEvaluatorBase.idl:

Canonical link: <a href="https://commits.webkit.org/260848@main">https://commits.webkit.org/260848@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0eb53819ee9417f4ef2a4bc12001a6247a081269

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/109666 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/18784 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/42409 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/87/builds/1145 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/118788 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/113562 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/20254 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/9984 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/101935 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/115417 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/15076 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/98309 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/43296 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/97050 "Passed tests") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/29961 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/85081 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/11515 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/70/builds/31303 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/12171 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/8239 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/17537 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/50912 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/7521 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/13915 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WebCore/xml/XPathEvaluator.h:
```diff
@@ -41,6 +41,7 @@ class XPathEvaluator : public RefCounted<XPathEvaluator> {
 
     ExceptionOr<Ref<XPathExpression>> createExpression(const String& expression, RefPtr<XPathNSResolver>&&);
     Ref<XPathNSResolver> createNSResolver(Node& nodeResolver);
+    void createNSResolverForBindings(Node&) const { } // Legacy.
```
I might suggest `static` instead of `const` since this is just to compile nothing.

cdumez replied:
Will fix, thanks.

darinadler commented on Source/WebCore/dom/Document.h:
```diff
@@ -1141,6 +1141,7 @@ class Document
     WEBCORE_EXPORT ExceptionOr<Ref<XPathExpression>> createExpression(const String& expression, RefPtr<XPathNSResolver>&&);
     WEBCORE_EXPORT Ref<XPathNSResolver> createNSResolver(Node& nodeResolver);
     WEBCORE_EXPORT ExceptionOr<Ref<XPathResult>> evaluate(const String& expression, Node& contextNode, RefPtr<XPathNSResolver>&&, unsigned short type, XPathResult*);
+    void createNSResolverForBindings(Node&) const { } // Legacy.
```
Same comment here.

cdumez committed:
Update Node.lookupNamespaceURI() and XPathEvaluatorBase.createNSResolver()
https://bugs.webkit.org/show_bug.cgi?id=252716

Reviewed by Darin Adler.

Update Node.lookupNamespaceURI() and XPathEvaluatorBase.createNSResolver() per:
https://github.com/whatwg/dom/pull/1165
https://github.com/web-platform-tests/wpt/pull/38636

- lookupNamespaceURI():
Element should handle "xml" and "xmlns" by default.

- createNSResolver():
It should return the argument as is, and should not add "xml" prefix support.

* LayoutTests/imported/w3c/web-platform-tests/dom/nodes/Node-lookupNamespaceURI-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/nodes/Node-lookupNamespaceURI.html:
* LayoutTests/imported/w3c/web-platform-tests/domxpath/xpathevaluatorbase-creatensresolver-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/domxpath/xpathevaluatorbase-creatensresolver.html: Added.
* Source/WebCore/dom/Document.h:
(WebCore::Document::createNSResolverForBindings const):
* Source/WebCore/dom/Node.cpp:
(WebCore::locateDefaultNamespace):
* Source/WebCore/xml/XPathEvaluator.h:
(WebCore::XPathEvaluator::createNSResolverForBindings const):
* Source/WebCore/xml/XPathEvaluatorBase.idl:

Canonical link: https://commits.webkit.org/260848@main
